discord-internal.h

```diff
@@ -430,6 +430,10 @@ struct discord_gateway_cbs {
   /** triggers when a thread is deleted */
   discord_on_channel on_thread_delete;
 
+  discord_on_guild on_guild_create;
```

Comment:
You could add some brief description here similar to the callbacks surrounding it, its up to you though!